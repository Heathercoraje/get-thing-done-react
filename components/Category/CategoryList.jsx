import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';


const Wrapper = styled.div`
	padding-left: 0.6rem;
	color: white;
`;
const CategoryList = ({ categories, ...props }) => (
	<Wrapper className="category-list">
		{categories.map(category => (
    <CategoryItem key={category.id} {...props} {...category} />
		))}
  </Wrapper>
);

CategoryList.propTypes = {
	props: PropTypes.object,
	categories: PropTypes.array,
	category: PropTypes.object,
	id: PropTypes.number
}
const CategoryItem = props => (
	<div className="category-div">
		<li
			onClick={props.click}
			value={props.id}
			data-name={props.name}
			className="category-text"
		>
			{props.name}
  </li>
		<button className="button-delete" onClick={props.delete} value={props.id}>
			&#8854;
  </button>
  </div>
);
CategoryItem.propTypes = {
	id: PropTypes.number,
	name: PropTypes.string,
	click: PropTypes.func,
	delete: PropTypes.func
}

export { CategoryList, CategoryItem };
