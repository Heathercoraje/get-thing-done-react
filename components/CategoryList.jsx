import React from 'react';
// they are just all function with return value
// which happens to be a component, simple like that

const CategoryList = ({ categories, deleteCategory, clickCate, selectedCategory }) => {
  const targetCategories = categories.slice(1);
  const Renderlist = categories =>
    targetCategories.map(category => (
      <div key={category.id}>
        <li onClick={clickCate} id={category.id}>
          {category.category}
        </li>
        <button onClick={deleteCategory} value={category.id}>
					Delete
        </button>
      </div>
    ));
  return (
    <div className="left">
      <li onClick={clickCate} id="0">
				Todo
      </li>
      <div>{Renderlist(categories)}</div>
    </div>
  );
};

export default CategoryList;
