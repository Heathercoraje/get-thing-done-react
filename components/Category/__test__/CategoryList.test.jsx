import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { CategoryList, CategoryItem } from '../CategoryList';

test('CategoryList should render as expected', () => {
	const categories = [
		{ id: 0, name: 'Tonight', description: 'Things to do tonight' }
	];
	const component = shallow(<CategoryList categories={categories} />);
	const tree = toJson(component);
	expect(tree).toMatchSnapshot();
	expect(component.length).toBe(1);
});

describe('CategoryList should render correctly', () => {
	it('should render CategoryItem', () => {
		const categories = [
			{ id: 0, name: 'Tonight', description: 'Things to do tonight' }
		];
		const component = shallow(<CategoryList categories={categories} />);
		expect(component.contains(  <CategoryItem
        description="Things to do tonight"
        id={0}
        key="0"
        name="Tonight"
      />)).toBe(true);
	});
});
