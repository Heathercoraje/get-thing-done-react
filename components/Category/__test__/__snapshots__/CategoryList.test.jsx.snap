// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList should render as expected 1`] = `
<styled.div
  className="category-list"
>
  <CategoryItem
    description="Things to do tonight"
    id={0}
    key="0"
    name="Tonight"
  />
</styled.div>
`;
